src/service.rs

```diff
@@ -938,17 +927,18 @@ impl Service {
     }
 
     /// Request an external node's ENR.
-    fn request_enr(
```

Comment:
If you want I can add the checks to the request_enr() in discv5.rs? 

I am finding 2 issues with doing what you stated above

1. Since request_enr() in service.rs isn't public how could other programs use it (compiling is giving me a deadcode warning)
2. if we were to call ``request_find_node_designated_peer()`` we would need to create a oneshot channel, but if we do that don't we have to do async move, but since request_enr() returns () that isn't possible?

Reply:
Oh sorry. This is my bad. 

I meant to point to the discv5.rs struct. I realize you kept the old function and have already done this. However, in the response, you do `v.remove(0)`. If a malicious user sends us back an empty response, the node will then panic I believe. 

Once we have a callback (in the request_enr() function in discv5.rs)  I think we should handle those cases gracefully (as was previously done at the service level). I.e If we get more ENRs than we expect, log a warning etc.

Reply:
Oh that makes sense I will get to work on that 👍

Reply:
Ok I pushed changes which I believe resolve your concerns. Let me know what you think.